Integration test failing on a newly created project

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-
- If this is a potential security issue, please read our security policy and contact us privately first: https://github.com/jhipster/generator-jhipster/security/policy
-->

##### **Overview of the issue**
I have created a new project (Monolith) and w/o changing a bit running ```mvn clean install``. As a result one of the IT tests is failing all the time with 403 error.
<!-- Explain the bug, if an error is being thrown a stack trace helps -->

```
2024-01-19T17:55:51.702+01:00  INFO 34528 --- [marlin-agent] [           main] c.s.r.a.m.w.r.MarlinAgentKafkaResourceIT : Started MarlinAgentKafkaResourceIT in 2.072 seconds (process running for 78.53)

MockHttpServletRequest:
      HTTP Method = POST
      Request URI = /api/marlin-agent-kafka/publish
       Parameters = {message=[value-produce]}
          Headers = []
             Body = null
    Session Attrs = {SPRING_SECURITY_CONTEXT=SecurityContextImpl [Authentication=UsernamePasswordAuthenticationToken [Principal=org.springframework.security.core.userdetails.User [Username=user, Password=[PROTECTED], Enabled=true, AccountNonExpired=true, CredentialsNonExpired=true, AccountNonLocked=true, Granted Authorities=[ROLE_USER]], Credentials=[PROTECTED], Authenticated=true, Details=null, Granted Authorities=[ROLE_USER]]]}

Handler:
             Type = null

Async:
    Async started = false
     Async result = null

Resolved Exception:
             Type = null

ModelAndView:
        View name = null
            Model = null

FlashMap:
       Attributes = null

MockHttpServletResponse:
           Status = 403
    Error message = Forbidden
          Headers = [Vary:"Origin", "Access-Control-Request-Method", "Access-Control-Request-Headers", Set-Cookie:"XSRF-TOKEN=432f553f-08ae-4127-99d9-1f04d39f64f8; Path=/", X-Content-Type-Options:"nosniff", X-XSS-Protection:"0", Cache-Control:"no-cache, no-store, max-age=0, must-revalidate", Pragma:"no-cache", Expires:"0", X-Frame-Options:"DENY"]
     Content type = null
             Body =
    Forwarded URL = null
   Redirected URL = null
          Cookies = [[Cookie@77d6fd21 name = 'XSRF-TOKEN', value = '432f553f-08ae-4127-99d9-1f04d39f64f8', comment = [null], domain = [null], maxAge = -1, path = '/', secure = false, version = 0, httpOnly = false]]
2024-01-19T17:55:51.751+01:00  WARN 34528 --- [marlin-agent] [           main] o.s.w.s.h.HandlerMappingIntrospector     : Cache miss for REQUEST dispatch to '/api/marlin-agent-kafka/register' (previous null). Performing MatchableHandlerMapping lookup. This is logged once only at WARN level, and every time at TRACE.
[ERROR] Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.324 s <<< FAILURE! -- in com.stukans.refirmware.agent.marlin.web.rest.MarlinAgentKafkaResourceIT
[ERROR] com.stukans.refirmware.agent.marlin.web.rest.MarlinAgentKafkaResourceIT.producesMessages -- Time elapsed: 0.026 s <<< FAILURE!
java.lang.AssertionError: Status expected:<200> but was:<403>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at com.stukans.refirmware.agent.marlin.web.rest.MarlinAgentKafkaResourceIT.producesMessages(MarlinAgentKafkaResourceIT.java:46)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

[INFO] Running com.stukans.refirmware.agent.marlin.web.rest.errors.ExceptionTranslatorIT

        ██╗ ██╗   ██╗ ████████╗ ███████╗   ██████╗ ████████╗ ████████╗ ███████╗
        ██║ ██║   ██║ ╚══██╔══╝ ██╔═══██╗ ██╔════╝ ╚══██╔══╝ ██╔═════╝ ██╔═══██╗
        ██║ ████████║    ██║    ███████╔╝ ╚█████╗     ██║    ██████╗   ███████╔╝
  ██╗   ██║ ██╔═══██║    ██║    ██╔════╝   ╚═══██╗    ██║    ██╔═══╝   ██╔══██║
  ╚██████╔╝ ██║   ██║ ████████╗ ██║       ██████╔╝    ██║    ████████╗ ██║  ╚██╗
   ╚═════╝  ╚═╝   ╚═╝ ╚═══════╝ ╚═╝       ╚═════╝     ╚═╝    ╚═══════╝ ╚═╝   ╚═╝

:: JHipster ?  :: Running Spring Boot 3.2.0 :: Startup profile(s) testdev ::
:: https://www.jhipster.tech ::

2024-01-19T17:55:52.031+01:00  INFO 34528 --- [marlin-agent] [           main] ContainersSpringContextCustomizerFactory : Warming up the kafka broker
2024-01-19T17:55:52.035+01:00  INFO 34528 --- [marlin-agent] [           main] c.s.r.a.m.w.r.e.ExceptionTranslatorIT    : Starting ExceptionTranslatorIT using Java 17.0.3.1 with PID 34528 (started by artjo in E:\Projects\JHipster\refirmware\test)
2024-01-19T17:55:52.036+01:00  INFO 34528 --- [marlin-agent] [           main] c.s.r.a.m.w.r.e.ExceptionTranslatorIT    : The following 1 profile is active: "testdev"
2024-01-19T17:55:54.060+01:00  INFO 34528 --- [marlin-agent] [           main] c.s.r.a.m.w.r.e.ExceptionTranslatorIT    : Started ExceptionTranslatorIT in 2.103 seconds (process running for 80.887)
2024-01-19T17:55:54.082+01:00  WARN 34528 --- [marlin-agent] [           main] o.s.w.s.h.HandlerMappingIntrospector     : Cache miss for REQUEST dispatch to '/api/exception-translator-test/missing-servlet-request-part' (previous null). Performing MatchableHandlerMapping lookup. This is logged once only at WARN level, and every time at TRACE.
2024-01-19T17:55:54.162+01:00  WARN 34528 --- [marlin-agent] [           main] o.s.web.servlet.PageNotFound             : Request method 'POST' is not supported
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.465 s -- in com.stukans.refirmware.agent.marlin.web.rest.errors.ExceptionTranslatorIT
[INFO]
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR]   MarlinAgentKafkaResourceIT.producesMessages:46 Status expected:<200> but was:<403>
[INFO]
[ERROR] Tests run: 23, Failures: 1, Errors: 0, Skipped: 0
[INFO]
[INFO]
[INFO] --- jacoco-maven-plugin:0.8.11:report-integration (post-integration-tests) @ marlin-agent ---
[INFO] Loading execution data file E:\Projects\JHipster\refirmware\test\target\jacoco-it.exec
[INFO] Analyzed bundle 'Marlin Agent' with 32 classes
[INFO]
[INFO] --- maven-checkstyle-plugin:3.3.1:check (default) @ marlin-agent ---
[INFO] You have 0 Checkstyle violations.
[INFO]
[INFO] --- maven-failsafe-plugin:3.2.2:verify (verify) @ marlin-agent ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:00 min
[INFO] Finished at: 2024-01-19T17:55:56+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:3.2.2:verify (verify) on project marlin-agent: There are test failures.
[ERROR]
[ERROR] Please refer to E:\Projects\JHipster\refirmware\test\target\failsafe-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

##### **Motivation for or Use Case**
A newly created project should not have failing tests.
<!-- Explain why this is a bug for you -->

##### **Reproduce the error**
create a project with the provided .yo-rc.json and run ```mvn clean install``` command
<!-- For bug reports, an unambiguous set of steps to reproduce the error -->

##### **Related issues**
Yes, I had a similar issue in the past with an older version of JHipster, but that bug was not solved.
<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

##### **JHipster Version(s)**
8.1.0
<!--
Which version of JHipster are you using, is it a regression?
-->

##### **JHipster configuration**
```
PS E:\Projects\JHipster\refirmware\test> jhipster info

        ██╗ ██╗   ██╗ ████████╗ ███████╗   ██████╗ ████████╗ ████████╗ ███████╗
        ██║ ██║   ██║ ╚══██╔══╝ ██╔═══██╗ ██╔════╝ ╚══██╔══╝ ██╔═════╝ ██╔═══██╗
        ██║ ████████║    ██║    ███████╔╝ ╚█████╗     ██║    ██████╗   ███████╔╝
  ██╗   ██║ ██╔═══██║    ██║    ██╔════╝   ╚═══██╗    ██║    ██╔═══╝   ██╔══██║
  ╚██████╔╝ ██║   ██║ ████████╗ ██║       ██████╔╝    ██║    ████████╗ ██║  ╚██╗
   ╚═════╝  ╚═╝   ╚═╝ ╚═══════╝ ╚═╝       ╚═════╝     ╚═╝    ╚═══════╝ ╚═╝   ╚═╝
                            https://www.jhipster.tech
Welcome to JHipster v8.1.0

Welcome to the JHipster Information Sub-Generator


```
marlin-agent@0.0.0 E:\Projects\JHipster\refirmware\test
`-- generator-jhipster@8.1.0
```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "authenticationType": "oauth2",
    "baseName": "marlin-agent",
    "buildTool": "maven",
    "cacheProvider": "no",
    "creationTimestamp": 1705655658027,
    "databaseType": "no",
    "devDatabaseType": "no",
    "enableGradleEnterprise": null,
    "enableHibernateCache": false,
    "enableSwaggerCodegen": false,
    "enableTranslation": false,
    "entities": [],
    "feignClient": false,
    "gradleEnterpriseHost": null,
    "jhipsterVersion": "8.1.0",
    "messageBroker": "kafka",
    "monorepository": true,
    "nativeLanguage": "en",
    "packageName": "com.stukans.refirmware.agent.marlin",
    "prodDatabaseType": "no",
    "reactive": false,
    "searchEngine": false,
    "serverPort": null,
    "serverSideOptions": [
      "messageBroker:kafka"
    ],
    "serviceDiscoveryType": false,
    "skipClient": true,
    "skipCommitHook": true,
    "skipUserManagement": true,
    "testFrameworks": [],
    "websocket": false
  }
}
</pre>
</details>


##### **Environment and Tools**

java version "17.0.3.1" 2022-04-22 LTS
Java(TM) SE Runtime Environment (build 17.0.3.1+2-LTS-6)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.3.1+2-LTS-6, mixed mode, sharing)

git version 2.40.1.windows.1

node: v20.11.0
npm: 10.2.4

Docker version 24.0.6, build ed223bc

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>
```

<!--
To provide all information we need, you should run `jhipster info` in the project root folder, and
copy/paste the result here.
The `.yo-rc.json` file generated in the root folder is mandatory for bug reports. This will help us to replicate the scenario.
You should remove any sensitive information like the rememberMe key or the jwtSecretKey key.
---
If you have a JDL please wrap it in below structure
  <details>
  <summary>JDL definitions</summary>
  <pre>
     JDL content here
  </pre>
  </details>
-->

[
[.yo-rc.json](https://github.com/jhipster/generator-jhipster/files/13992078/default.yo-rc.json)
](url)

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<!--
If the error is during an entity creation or associated with a specific entity.
If you are using JDL, please share that configuration as well.
-->

##### **Browsers and Operating System**
Windows 10 PRO [Version 10.0.19045 Build 19045]
Chrome Version 120.0.6099.225 (Official Build) (64-bit)

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

- [ X] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/spring-cloud-stream/templates/src/test/java/_package_/web/rest/KafkaResourceIT_imperative.java.ejs
+++ b/generators/spring-cloud-stream/templates/src/test/java/_package_/web/rest/KafkaResourceIT_imperative.java.ejs
@@ -23,6 +23,7 @@ import <%= packageName %>.config.EmbeddedKafka;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.fail;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
@@ -62,7 +63,7 @@ class <%= upperFirstCamelCaseBaseName %>KafkaResourceIT {
 
     @Test
     void producesMessages() throws Exception {
-        restMockMvc.perform(post("/api/<%= dasherizedBaseName %>-kafka/publish?message=value-produce")).andExpect(status().isOk());
+        restMockMvc.perform(post("/api/<%= dasherizedBaseName %>-kafka/publish?message=value-produce").with(csrf())).andExpect(status().isOk());
         assertThat(output.receive(1000, "binding-out-0").getPayload()).isEqualTo("value-produce".getBytes());
     }
 

--- a/generators/spring-cloud-stream/templates/src/test/java/_package_/web/rest/KafkaResourceIT_imperative.java.ejs
+++ b/generators/spring-cloud-stream/templates/src/test/java/_package_/web/rest/KafkaResourceIT_imperative.java.ejs
@@ -23,7 +23,9 @@ import <%= packageName %>.config.EmbeddedKafka;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.fail;
+<%_ if (authenticationUsesCsrf) { _%>
 import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+<%_ } _%>
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
@@ -63,7 +65,7 @@ class <%= upperFirstCamelCaseBaseName %>KafkaResourceIT {
 
     @Test
     void producesMessages() throws Exception {
-        restMockMvc.perform(post("/api/<%= dasherizedBaseName %>-kafka/publish?message=value-produce").with(csrf())).andExpect(status().isOk());
+        restMockMvc.perform(post("/api/<%= dasherizedBaseName %>-kafka/publish?message=value-produce")<% if (authenticationUsesCsrf) { %>.with(csrf())<% } %>).andExpect(status().isOk());
         assertThat(output.receive(1000, "binding-out-0").getPayload()).isEqualTo("value-produce".getBytes());
     }
 

